pages/src/App.js

```diff
@@ -1,80 +1,62 @@
 import React, { Component } from 'react'
 import { Map } from 'immutable'
-
+import { provideState, update, mergeIntoState } from "freactal";
 import logo from './logo.svg'
 import TodoList from './TodoList.js'
 import './App.css'
-
-class App extends Component {
-  state = {
-    todos: new Map()
-  }
-
-  componentWillMount () {
-    fetch('//localhost:20431/todos').then(res =>
-      res.json().then(todos => {
-        this.setState({
+import "isomorphic-fetch";
+
+const wrapComponentWithState = provideState({
+  initialState: () => ({
+    todos: new Map(),
+    newTodoLabel: '',
+    gitInfo: ''
+  }),
+  effects: {
+    initialize: (effects) => fetch('//localhost:20431/todos')
+      .then(res => res.json()
+        .then(todos => mergeIntoState({
           todos: new Map().withMutations(map => {
             todos.forEach(todo => {
               map.set(todo.id, todo)
             })
           })
         })
-      })
-    )
-  }
-
-  _onCreateTodo = todo => {
-    const { id } = todo
-    const { todos } = this.state
-    const previous = todos.get(id)
-
-    this.setState({ todos: todos.set(id, todo) })
-
-    fetch(`//localhost:20431/todos/${id}`, {
+        )),
+    sendToServer: (effects, todo, id) => fetch(`http://localhost:20431/todos/${id}`, {
```

Comment:
You do not need an `id` param, it's already in the `todo`.

Reply:
copy that